app/controllers/users.rb

```diff
@@ -0,0 +1,20 @@
+get '/users/new' do
+  @user = User.new
+  erb :"users/new"
+end
+
+post '/users' do
+  @user = User.new(:first_name => params[:first_name],
+                    :last_name => params[:last_name],
+                    :username => params[:username],
+                    :email => params[:email],
+                    :password => params[:password],
+                    :password_confirmation => params[:password_confirmation])
```

Comment:
Could this extraction be dealt with in the model? i.e. pass params directly through and create a tested method in the model